### PR TITLE
[UM-20]-created-student-report

### PR DESCRIPTION
**Pull request by yncrahmet**
konsol

### CODE REVIEW

File: src/main/java/com/archisacadeny/student/StudentRepository.java [hunk 1]
```diff
@@ -65,4 +67,65 @@ public Student createStudent(Student student) {
...
+
+                long studentIdFromDb = resultSet.getLong("Id");
+                String studentName = resultSet.getString("FullName");
+
+                List<CourseGrade> courseGrades = new ArrayList<>();
+                int totalGradePoints = 0;
+                int totalCredits = 0;
+                double gpa = 0.0;
+
+                do {
```
Comment (JamshidS) [inline src/main/java/com/archisacadeny/student/StudentRepository.java]: you don't need to create a sperate table for storing grades and etc it is already there in Oruj's PR please do take a look together

**Review by JamshidS — CHANGES_REQUESTED** [review]

File: src/main/java/com/archisacadeny/student/CourseStudentMapper.java [hunk 2]
```diff
@@ -29,11 +30,12 @@ CONSTRAINT fk_student_id FOREIGN KEY (student_id) REFERENCES "public"."persons"(
         }
     }
 
-    public static void saveToCourseStudentMapper(int studentID,int courseId){
-        String query = "INSERT INTO course_student_mapper(student_id,course_id) VALUES(?,?)";
+    public static void saveToCourseStudentMapper(int studentID,int courseId, double grade){
```
Comment (JamshidS) [inline src/main/java/com/archisacadeny/student/CourseStudentMapper.java]:
```suggestion
    public void saveToCourseStudentMapper(int studentID,int courseId, double grade){
```

File: src/main/java/com/archisacadeny/student/StudentRepository.java [hunk 3]
```diff
@@ -65,4 +65,57 @@ public Student createStudent(Student student) {
 
         return student;
     }
+
+    public StudentReport generateStudentAchievementReport(int studentId) {
```
Comment (JamshidS) [inline src/main/java/com/archisacadeny/student/StudentRepository.java]:
```suggestion
    public Map<String,Object> generateStudentAchievementReport(int studentId) {
```

File: src/main/java/com/archisacadeny/student/StudentRepository.java [hunk 4]
```diff
@@ -65,4 +65,57 @@ public Student createStudent(Student student) {
...
+
+                do {
+                    long courseId = resultSet.getLong("course_id");
+                    int grade = resultSet.getInt("grade");
+                    int courseCredit = getCourseCredit(courseId);
+                    totalWeightedGradePoints += grade * courseCredit;
+                    totalCredits += courseCredit;
+                } while (resultSet.next());
+
+                double gpa = totalCredits > 0 ? totalWeightedGradePoints / totalCredits : 0.0;
```
Comment (JamshidS) [inline src/main/java/com/archisacadeny/student/StudentRepository.java]: business logic should be inside the service class not repository please do update it accordingly

**Review by JamshidS — CHANGES_REQUESTED** [review]

**Comment by yncrahmet** [comment]
çıktı

**Review by JamshidS — APPROVED** [review]

**Pull request merged**